### PR TITLE
Lpabon/6140 inline value sets

### PR DESCRIPTION
**Pull request by TurkeyFried**
This PR adds support for Value Sets in the FhirToHL7 translation config. This replaces various custom functions that had been built as a workaround.

Test Steps:
1. Run the converter with the fhirdata command. e.g. `./prime fhirdata --input-file src/testIntegration/resources/datatests/HL7_to_FHIR/sample_co_1_20220518-0001.fhir -s metadata/hl7_mapping/ORU_R01/ORU_R01-base.yml --output-format=HL7`
2. Search on the config files for `valueSet` and verify that the values map correctly. For example, `OBR-25` (result status) is configured to map certain strings to single character values.

## Changes
- Added support for Value Sets on YML configs.
- Replaced custom FHIR functions that existed just for mapping with Value Sets instead.
- Added test for getting mapped values.

## Linked Issues
- #6140

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt [hunk 1]
```diff
@@ -152,7 +152,8 @@ class FhirToHl7Converter(
         var retVal = ""
         run findValue@{
             element.value.forEach {
-                val value = if (it.isBlank()) ""
+                val value = if(element.valueSet.isNotEmpty()) it
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt:155]: I am not sure what this is doing.  Can you explain? `if(element.valueSet.isNotEmpty()) it`

File: prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml [hunk 2]
```diff
@@ -30,8 +30,12 @@ elements:
 
   - name: xad-use
     condition: '%resource.use.exists()'
-    value: [ '%resource.use.value.getAddressUse()' ]
+    value: [ '%resource.use.value' ]
     hl7Spec: [ '%{hl7XADField}-7' ]
+    valueSet:
+      home: H
```
Comment (carlosfelix2) [inline prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml:36]: Add the rest of the values that we know we can convert.  We want to start being as complete as we can to support any kind of data even if we do not have sample data for it.

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt [hunk 3]
```diff
@@ -207,16 +208,22 @@ data class ConfigSchemaElement(
...
             schema.isNullOrBlank() && hl7Spec.isEmpty() ->
                 addError("Hl7Spec property is required when not using a schema")
             schema.isNullOrBlank() && value.isEmpty() ->
                 addError("Value property is required when not using a schema")
         }
 
+        // value sets need a value to be...set
+        when {
+            valueSet.isNotEmpty() && value.isEmpty() ->
+                addError("Value property is required when using a value set")
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/schema/ConfigSchema.kt:224]: nice!

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt [hunk 4]
```diff
@@ -103,17 +95,10 @@ object ConstantSubstitutor {
  * to the function name in the CustomFHIRFunctions class
  */
 enum class CustomFHIRFunctionNames {
-    GetAddressUse,
     GetPhoneNumberCountryCode,
     GetPhoneNumberAreaCode,
     GetPhoneNumberLocalNumber,
     GetPhoneNumberExtension,
-    GetTelecomUseCode,
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt:111]: LOVE IT!

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (TurkeyFried) [reply prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml:36]: Found a few more values that were easy to map. there's a lot of cases where we will need more complicated logic, so those are not included

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt (hunk 1 above)
Comment (TurkeyFried) [reply prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt]: Reverted this as it was a bad way to solve an issue I had with a test

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt [hunk 5]
```diff
@@ -166,6 +166,11 @@ class FhirToHl7Converter(
                 }
             }
         }
+
+        // when valueSet is available, use the matching value else just pass the value as is
+        // does a lowerCase comparison
+        if (element.valueSet.isNotEmpty()) retVal = element.valueSet.getOrDefault(retVal.lowercase(), retVal)
```
Comment (carlosfelix2) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt:172]: Will this create problems if the value set is written in uppercase and the values come in as lowecase?

**Review by carlosfelix2 — CHANGES_REQUESTED** [review]

File: prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml (hunk 2 above)
Comment (TurkeyFried) [reply prime-router/metadata/hl7_mapping/ORU_R01/base/datatype/xad-extended-address.yml:36]: done!

**Review by carlosfelix2 — APPROVED** [review]

**Pull request merged**